File: AD-HOC/3048/main.dart
import 'dart:io';

void main() {
  int maxSequence = 1;
  int n = int.parse(stdin.readLineSync()!);
  String previous = stdin.readLineSync()!;
  for (int i = 1; i < n; i++) {
    String next = stdin.readLineSync()!;
    if (previous != next) maxSequence++;
    previous = next;
  }
  print(maxSequence);
}

File: AD-HOC/3048/main.c
#include <stdio.h>

int main() {
    int n, previous, next, maxSequence = 1;
    scanf("%d %d", &n, &previous);
    for (int i = 1; i < n; i++) {
        scanf("%d", &next);
        if (previous != next) maxSequence++;
        previous = next;
    }
    printf("%d\n", maxSequence);
    return 0;
}
